runner/operator/roslin_operator/bin/retrieve_samples_by_query.py

```diff
@@ -32,6 +45,169 @@ def get_samples_from_patient_id(patient_id):
+    can't be sent as a value, so had to make a semi-redundant function
+    """
+    data_query_set = [Q(filemetadata__metadata__preservation=value) for value in set(data)]
+    query = data_query_set.pop()
+    for item in data_query_set:
+        query |= item
+    return query
+
+
+def get_pooled_normals(run_ids, preservation_types, bait_set):
+    """
+    From a list of run_ids, preservation types, and bait sets, get all potential pooled normals
+    """
+    file_objs = File.objects.prefetch_related(
+        Prefetch('filemetadata_set', queryset=FileMetadata.objects.select_related('file').\
+            order_by('-created_date')))
+    query = Q(file_group=settings.POOLED_NORMAL_FILE_GROUP)
+    run_id_query = Q()
+    preservation_query = Q()
+
+    run_id_query = build_run_id_query(run_ids)
+    preservation_query = build_preservation_query(preservation_types)
+    pooled_normals = file_objs.\
+        filter(query & run_id_query & preservation_query).\
+        order_by('file_name')
+    descriptor = get_descriptor(bait_set, pooled_normals)
+
+    if not descriptor: # i.e., no pooled normal
+        return None
+    pooled_normals = pooled_normals.filter(filemetadata__metadata__recipe=descriptor)
+    sample_files = list()
+
+    # arbitrarily named
+    sample_name = "pooled_normal_%s_%s_%s" % (descriptor,
+                                              "_".join(run_ids),
+                                              "_".join(preservation_types))
+
+    num_of_pooled_normals = len(pooled_normals)
+    if num_of_pooled_normals > 0:
+        for pooled_normal in pooled_normals:
+            sample = dict()
+            sample['id'] = pooled_normal.id
+            sample['path'] = pooled_normal.path
+            sample['file_name'] = pooled_normal.file_name
+            metadata = init_metadata()
+            metadata['sampleId'] = sample_name
+            metadata['sampleName'] = sample_name
+            metadata['requestId'] = sample_name
+            metadata['baitSet'] = descriptor
+            metadata['recipe'] = descriptor
+            metadata['run_id'] = run_ids
+            metadata['preservation'] = preservation_types
+            # because rgid depends on flowCellId and barcodeIndex, we will
+            # spoof barcodeIndex so that pairing can work properly; see
+            # build_sample in runner.operator.roslin_operator.bin
+            metadata['R'] = get_r_orientation(pooled_normal.file_name)
+            metadata['barcodeIndex'] = spoof_barcode(sample['file_name'], metadata['R'])
+            metadata['flowCellId'] = 'PN_FCID'
+            metadata['tumorOrNormal'] = 'Normal'
+            metadata['patientId'] = 'PN_PATIENT_ID'
+            sample['metadata'] = metadata
+            sample_files.append(sample)
+        pooled_normal = build_sample(sample_files, ignore_sample_formatting=True)
+        return pooled_normal
+    return None
+
+
+def get_r_orientation(fastq_filename):
```

Comment:
Just realized this is crappy.

Will change this to what's in the ETL logic real quick.